interface/requests/wireheader.go

```diff
@@ -6,68 +6,165 @@ package requests
 import (
 	"bytes"
 	"encoding/binary"
-	"errors"
+	"fmt"
 
-	"github.com/sirupsen/logrus"
+	"github.com/parallaxsecond/parsec-client-go/interface/auth"
 )
 
-// WireHeader represents a request header
-type WireHeader struct {
+type contentType uint8
+type versionMajorType uint8
+type versionMinorType uint8
+type flagsType uint16
+type sessionType uint64
+type acceptType uint8
+
+const (
+	versionMajorOne  versionMajorType = 0x01
+	versionMinorZero versionMinorType = 0x00
+
+	flagsZero flagsType = 0x0000
+)
+
+func (f flagsType) isValid() bool {
+	return f == flagsZero
+}
+
+func (a acceptType) isValid() bool {
+	return a == acceptType(0x00) //nolint:gomnd // Only to be set on sent headers
+}
+
+const (
+	contentTypeProtobuf contentType = 0x00
+)
+
+func (c contentType) isValid() bool {
+	return c == contentTypeProtobuf
+}
+
+// magic number to indicate we have a wire header
+const magicNumber uint32 = 0x5EC0A710
+
+// wireHeader represents a request header
+type wireHeader struct {
 	magicNumber  uint32
 	hdrSize      uint16
-	versionMajor uint8
-	versionMinor uint8
-	Flags        uint16
-	Provider     uint8
-	Session      uint64
-	ContentType  uint8
-	AcceptType   uint8
-	AuthType     uint8
-	BodyLen      uint32
-	AuthLen      uint16
-	OpCode       uint32
-	Status       uint16
-	Reserved1    uint8
-	Reserved2    uint8
+	versionMajor versionMajorType
+	versionMinor versionMinorType
```

Comment:
Note that the first four fields:
- magic number
- header size
- major version number
- minor version number

will always have the same width and position on every Parsec request/response header, no matter the wire protocol version. It would then make sense to either separate them in their own structure or not include them in the `wireHeader` structure, which will be the one that is versioned. In the [Rust interface](https://github.com/parallaxsecond/parsec-interface-rs/blob/master/src/requests/common/wire_header_1_0.rs), we put the rest in its own `wire_header_1_0` module to make it easy to write the logic selecting the correct wire header format. Though we did not implement that logic yet, at least we have the building blocks 😋 you have some more info [here](https://github.com/parallaxsecond/parsec/issues/169)

Reply:
created this issue: https://github.com/parallaxsecond/parsec-client-go/issues/31